site/app/controllers/student/SubmissionController.php

```diff
@@ -1391,6 +1392,142 @@ private function updateSubmissionVersion() {
         return array('error' => false, 'version' => $new_version, 'message' => $msg);
     }
     
+
+	// This copy all files from given dir ($src) and its subdirs into one destination folder, $dst
+	private function recursive_copy($src, $dst) {
+ 		$dir = opendir($src);
+        while(false !== ( $file = readdir($dir)) ) {
+            if (( $file != '.' ) && ( $file != '..' ) && ( $file != '__MACOSX' )) {
+                if ( is_dir($src . '/' . $file) ) {
+                    $this->recursive_copy($src . '/' . $file, $dst);
+                }
+                else {
+                    $extension = strtolower(pathinfo($file, PATHINFO_EXTENSION));
+                    if ($extension === 'png') // just get all png file only
+                        copy($src . '/' . $file,$dst . '/' . $file);
+                }
+            }
+        }
+        closedir($dir);
+    }
```

Comment:
Move this function to the `FileUtils` library, since we can then more easily write tests for it, and at that time we can rewrite it to use the DirectoryIterator as well.